src/main/java/org/springframework/data/neo4j/repository/query/QueryFragmentsAndParameters.java

```diff
@@ -260,6 +275,33 @@ public boolean isScalarValueReturn() {
 			return orderBy != null ? orderBy : new SortItem[]{};
 		}
 
+		public Statement toGenericStatement() {
```

Comment:
Maybe something like "generateGenericStatement" conveys what's happening better. And maybe a comment for me.

Also: Does it need to be public?